set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -855,5 +860,37 @@ private void setSerializedItemIds(UserSet userSet) {
 	((WebUserSetImpl) userSet).setSerializedItems(jsonSerialized);
 //	userSet.setItems(null);
     }
+    
+    /*
+     * (non-Javadoc)
+     * 
+     * @see eu.europeana.UserSet.web.service.UserSetService#storeUserSet(eu.
+     * europeana.UserSet.definitions.model.UserSet, boolean)
+     */
+//    @Override
+    public UserSet updateUserSet(PersistentUserSet persistentUserSet, UserSet webUserSet) throws HttpException {
+    mergeUserSetProperties(persistentUserSet, webUserSet);
+    // update modified date
+    persistentUserSet.setModified(new Date());
+    updateTotal(persistentUserSet);
+    checkDuplicateUserSets(persistentUserSet, true);
```

Comment:
this check applies only to entity best items sets and it is a validation. Therefore it should be called at the appropriate place within the BaseUserSetServiceImpl.validateEntityBestItemsSet()

Reply:
Makes sense. I moved it there.